llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -546,8 +556,11 @@ def load_data(
         if not sharepoint_folder_id:
             sharepoint_folder_id = self.sharepoint_folder_id
 
+        if not sharepoint_site_id:
```

Comment:
`load_data` should take a new parameter `sharepoint_site_id` I think?

Reply:
just moved the check to line 560 now, keepin only on the initialization to not pass site_id all the way down